### DISCUSSION TITLE
Trying to wrap up a mid-term project based on data, but not vision data to predict inflation with what we learned.

### DISCUSSION THREAD

**Discussion by AxisMeetsWorld** [mrdbourke/pytorch-deep-learning#598] [Show and tell]
Hi @mrdbourke,
      I am trying to put together a project based on some research I did reading about the primary causes of inflation (*here in the US*) and even though there might be other factors, I learned that the primary influences seem to be GDP, number of jobs to people, and free money in the economy. I haven't quite completed it, but you did mention you might be interested in looking at it. ❓ I am about 2/3 of the way done with the project and wondered if it was better to send a link here, or if there is an email that I can share the work I have done thus far with you at?❓ 
❗ **I am really curious to get your perspective on the work and see if there is anything you might improve because I was hoping to post this eventually on my LinkedIn page, and your feedback would be highly valued.** ❗ Here are some preliminary plots I have regarding the data:
![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/29417743/b793dcf2-dbe4-485b-902c-8ae69283a53c)
Would like to get your perspective.

**Comment by AxisMeetsWorld** [comment]
@mrdbourke hopefully I can share my repository with you sometime soon. I figured that I needed to run regression analysis on this, and I am just not piecing together very well what you have taught to reduce the loss. Would love to figure this one out and share if you can help me figure out what I am doing wrong. Accuracy is less than 20% on almost all epochs. **One thing that I have been reading about, and you discussed in Chapt 4 was the concept of transforms. The only thing that gives me pause here is that most of the examples I have seen revolve around image data. I was able to find one or two examples that *I think* will work with flat data like what I am trying to use, but I cannot be sure.** I have the feeling that my dataset is too small right now, so it seems data augmentation might be a good option.

**Comment by mrdbourke** [comment]
Hi @AxisMeetsWorld , 

This looks like you're doing a time series analysis project.

And while it's similar to what's taught in the PyTorch course, many of the image-based techniques won't necessarily help.

--

To try and understand your problem better, what's an example of your ideal inputs and outputs of the problem? 

If it's time series analysis/forecasting you're trying to do, I'd suggest looking at the following time series libraries: 

* Facebook Kats - https://github.com/facebookresearch/Kats
* Salesforce Merlion - https://github.com/salesforce/Merlion
* Darts - https://github.com/unit8co/darts

**Reply by AxisMeetsWorld** [reply]
Hi @mrdbourke ,
           I will take a look at these. To answer the question, what I was trying to do was take what I had learned here in the course and apply it to work I do a little more regularly (alot of it is with societal/governmental data analysis). I realized it was a little stretch, but I was hoping to use the techniques here with a different dataset (and see if I could accommodate not having as much data available as you would get with an image). Further, there isn't as much memory I have available that many firms that would use image processing might have. 
           Ultimately, what I was hoping to do, was accommodate for lag times, and take the 3 input variables above to account for inflation metrics like the CPI (consumer price index), and train a neural network on those. Then, I was hoping to finish up by trying out the experimentation on future results. After getting this feedback, and some from a few friends, it seems what I was hoping to do just doesn't work with PyTorch. If you have any ideas though, besides looking at those packages, please let me know. Thanks!

**Reply by AxisMeetsWorld** [reply]
Also, I finished this very thorough course! I am going to have more time now to look over these other GitHub packages to see how they can help me produce this study, but if you want to see, I tried to create a summary of the many things learned here in your course! Here it is on LinkedIn: https://www.linkedin.com/feed/update/urn:li:share:7103170171216830465/
